include/deal.II/base/vectorization.h

```diff
@@ -167,9 +167,35 @@ struct EnableIfScalar<VectorizedArray<Number>>
  * similar to std::vector otherwise but always makes sure that data is
  * correctly aligned.
  *
+ * The user can explicitly control the used instruction set architecture (ISA)
+ *  extension by specifying the vector width/number of lanes via  the second
+ * template parameter of this wrapper class. For example on Intel Skylake,
+ * you have the following options for the data type double:
+ *  - VectorizedArray<double, 1> // no vectorization (auto-optimization)
+ *  - VectorizedArray<double, 2> // SSE2
+ *  - VectorizedArray<double, 4> // AVX2
```

Comment:
```suggestion
 *  - VectorizedArray<double, 4> // AVX
```
The width has been introduced by AVX two years before AVX2 and we should honor the main innovation, which was AVX. We hardly use AVX2-specific stuff, only for gather and implicitly via the compiler to transform MUL + ADD into FMA. And for the latter, Intel typically considers FMA a separate feature besides AVX2.